### DISCUSSION TITLE
How to access the database? Best way to track multiple channels?

### DISCUSSION THREAD

**Discussion by Andrej-VB** [chylex/Discord-History-Tracker#194] [General]
Wondering how I should access the database (preferably in a coding language such as Python.) I love the native GUI of the DHT but I'd like to access those values to use them elsewhere.

I should note I'd use DHT primarily for the Tracking feature, and I'm also curious what's the best way to track multiple channels and servers.

**Comment by chylex** [comment]
> Wondering how I should access the database (preferably in a coding language such as Python.) I love the native GUI of the DHT but I'd like to access those values to use them elsewhere.

I use [DataGrip](https://www.jetbrains.com/datagrip/), but that's a paid program. For programming languages, any language that has bindings for SQLite 3.35.5 or newer will work; Python seems to have [native support](https://docs.python.org/3/library/sqlite3.html).

> what's the best way to track multiple channels and servers

You can copy the tracking script into multiple browser windows (multiple tabs might also work, but I found that if having a Discord tab in the background caused tracking to slow down or stop).

The easiest would be to have one browser window per server, and use the option to switch to the next channel when reaching the first message in a channel. You could also try one browser window per channel, if there are few very long channels, but then I wouldn't recommend automatically switching to the next channel - I don't know what can go wrong if two scripts try to track the same channel at once.

**Reply by Andrej-VB** [reply]
I've had no trouble navigating through Python's native support with the docs you linked, thank you! I was confused at first since I've had trouble opening the .dht file in Dbeaver and MySQL but this works great! 

I haven't yet tried the multi-browser setup, only ran the code inside the discord app, but I feel like there shouldn't be any issues. Can these browsers save into the same database or is it better to keep them separate?

**Reply by chylex** [reply]
Organize your databases however it makes sense for you, recent versions of DHT can have many tracking scripts all saving to the same database with just one instance of the app open.
